druid/src/widget/viewport.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2020 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::kurbo::{Affine, Point, Rect, Size, Vec2};
+use crate::widget::prelude::*;
+use crate::{Data, WidgetPod};
+
+/// Represents the size and position of a rectangular "viewport" into a larger area.
+#[derive(Clone, Copy, Default, Debug, PartialEq)]
+pub struct RectPort {
```

Comment:
I feel like there must be a better name than this but I can't think of one right now...

Reply:
We could call *this* `Viewport` and call the widget `ClipContainer` or something? just spitballing...

Reply:
Viewport is rapidly becoming a very overloaded term. In some cases it can be the size of the window, in other situations it refers to the paint rect of a widget, and now we are adding two new things which each reference related topics with similar terminology. I don't have much against `RectPort` but I do like renaming the widget to `ClipContainer` or something else.

Reply:
Hmm, `ClipContainer` is missing half the story for me, since I see this widget as being about both clipping and panning. I'm going to try sticking with `Viewport` for now and renaming `RectPort` to `ViewportPos`...

Reply:
Can you clarify how the distinction between clipping and panning? I think of panning as just a particular subjective experience of something being clipped. 🤔

Reply:
I don't really have a *very* strong opinion on this, but when I hear "clip" I just think of the graphical operation. Like, even without a transformation, a `ClipContainer` might be able to clip a child so that it has rounded corners or something? That does admittedly sound pretty contrived...

Reply:
Cool, just curious. I think it's fairly common for UI toolkits to call this sort of thing a 'clip view'; at least, that's what it's [called in AppKit](https://developer.apple.com/documentation/appkit/nsclipview).